pkg/framework/plugins/removepodsviolatingtopologyspreadconstraint/main_test.go

```diff
@@ -0,0 +1,12 @@
+package removepodsviolatingtopologyspreadconstraint
+
+import (
+	"context"
+	"testing"
+
+	"sigs.k8s.io/descheduler/pkg/tracing"
+)
+
+func TestMain(m *testing.M) {
```

Comment:
@damemi  @a7i  Should I take this route or add a simple `init` handler on `pkg/tracing` and set the provider to No Op and create a tracer out of it by default that can later be overwritten ?

Reply:
@damemi  That might get a bit tricky. 

```golang
func SetTracerProvider(tp trace.TracerProvider) {
	current := TracerProvider()

	if _, cOk := current.(*tracerProvider); cOk {
		if _, tpOk := tp.(*tracerProvider); tpOk && current == tp {
			// Do not assign the default delegating TracerProvider to delegate
			// to itself.
			Error(
				errors.New("no delegate configured in tracer provider"),
				"Setting tracer provider to it's current value. No delegate will be configured",
			)
			return
		}
	}

	delegateTraceOnce.Do(func() {
		if def, ok := current.(*tracerProvider); ok {
			def.setDelegate(tp)
		}
	})
	globalTracer.Store(tracerProviderHolder{tp: tp})
}
```

So, on the actual runtime the delegate settings might not be configured for the actual trace. Is there a better way to handle this ?

Reply:
Only way I can think of is using an env lookup in the `init` and do this conditionally and set that ENV during the tests

```golang
func init() {
   if v, ok := os.LookupEnv("INIT_NOOP_TRACER"); ok && v == "true" {
      // init provider and tracer
   }
}
```
But I am not sure if this is the right thing to do or not.

Reply:
something like:

```
var tracer

init() {
  tracer = otel.NoOpTracer()
}

SetTracerProvider() {
  opts := parseOptions()
  tracer = otel.SetTracerProvider()...
}
```

Reply:
@damemi  changed the `init` handler to avoid calling `otel.SetTraceProvider` entirely. 

```golang
func init() {
	provider = trace.NewNoopTracerProvider()
	tracer = provider.Tracer(TracerName)
}
```
When imported, it will have the tracer and provider set to NoOp version of it. But it will get overwritten by the right provider and tracer at the runtime and then we call the `otel.SetTraceprovider` so it will take care of the required runtime config at the OTEL level.